doc/deployment.md

```diff
@@ -0,0 +1,51 @@
+# Production Deployment
+
+1. Ensure changelog is up to date
+- Do a commit log diff against the previous version, ensure all the important changes are in the upcoming release
+- Keep a blank upcoming release
+- Verify changelog with other teams
+ 
+2. Tag release
+- Create a tag off master with the same name as in the changelog. Typically with the format: yyyy-mm-dd-n.
+- Add a short description in the tag for the release
+- `git tag -a <yyyy-mm-dd-n> -m "<description>"`
+- `git push --tags`
+
+3. Understand migrations to be run as a part of this release
+- `git diff <previous-tag>..<current-tag> db/migrate`
+- lookout for data migrations, and irreversible migrations   
+
+Perform the following steps first on staging, QA it, and then perform them on production.
+
+4. If there is a data migration
+- Check for changes in lib: `git diff -w 2018-08-13-1..2018-09-17-1 lib/`
```

Comment:
This diff tells us if there are any data migrations, or important scripts/seeds that need to be run. The following steps just say "look out for" migrations, etc because what needs to be done depends on the context.

Sometimes, data migrations can be run after the deploy, and sometimes they need to be run before. The `cap deploy` script does a code deploy, runs schema migrations, and starts the service. We might need to modify it to not start the service until the data migrations are run.

We decided we won't detail this, or modify the scripts until we come across such a scenario. But it's probably worth adding the above description into the document.

Thanks for pointing this out, @timcheadle :)